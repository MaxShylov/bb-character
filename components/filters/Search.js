import { useCallback, useState } from 'react';
import { StyleSheet, TextInput, View, Button } from 'react-native';

export const Search = ({ onSearch }) => {
  const [search, setSearch] = useState();
  const handleSearch = useCallback(() => onSearch(search), [search])

  return (
    <View style={styles.wrapper}>
      <TextInput
        style={styles.input}
        placeholder="Search by name"
        onChangeText={setSearch}
        onSubmitEditing={handleSearch}
      />
      <Button title="Search" onPress={handleSearch} />
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderWidth: 1,
    marginVertical: 12,
    paddingHorizontal: 10,
  },
  input: {
    height: 50,
    flexGrow: 1,
    fontSize: 16,
  },
});
