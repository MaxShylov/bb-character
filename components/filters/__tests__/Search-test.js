import renderer from 'react-test-renderer';
import { Search } from '../Search';

test('renders correctly', () => {
  const tree = renderer.create(<Search />).toJSON();

  expect(tree).toMatchSnapshot();
});
