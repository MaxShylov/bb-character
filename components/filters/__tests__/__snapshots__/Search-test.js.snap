// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 12,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Search by name"
    style={
      Object {
        "flexGrow": 1,
        "fontSize": 16,
        "height": 50,
      }
    }
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Search
      </Text>
    </View>
  </View>
</View>
`;
